File: src/test/java/TestData/TransactionData.java
package TestData;

import java.util.HashMap;
import java.util.Map;

public class TransactionData {
    public Map<Object, Object> authorizationData(float amount, String cardToken){
        Map<Object, Object> transactionData = new HashMap<>();
        transactionData.put("network","VISA");
        transactionData.put("card_token", cardToken);
        transactionData.put("amount", amount);
        transactionData.put("settlement_date", "2019-02-23");
        transactionData.put("mti", "0100");
        return transactionData;
    }

    public Map<Object, Object> transactionClearingData( float amount, String cardToken){
        Map<Object, Object> transactionData = new HashMap<>();
        transactionData.put("network","VISA");
        transactionData.put("card_token", cardToken);
        transactionData.put("amount", amount);
        transactionData.put("reason_code", 9000);
        return transactionData;
    }
}

File: ruby/lib/test-data/transaction_data.rb

module TransactionData

  @base_clearing = lambda {
     {
      network: 'VISA',
      card_token: 'card_token',
      amount: 'amount',
      network_reference_id: 'network_reference_id',
      acquirer_reference_id: 'acquirer_reference_id',
      settlement_date: '2019-02-23',
      reason_code: 9000
  }
  }

  def clear_visa_transaction(amount, card_token, network_ref_id, acquirer_ref_id, headers)
    @modified_clearing = @base_clearing.call.clone
    @modified_clearing.update({
                                  amount: amount,
                                  card_token: card_token,
                                  network_reference_id: network_ref_id,
                                  acquirer_reference_id: acquirer_ref_id
                        })
    JSON.parse(RestClient.post("#{Config.jcard_url}/simulate/advanced/clearing", @modified_clearing.to_json, headers))
  end
end
